Add support for computing the inverse hyperbolic cotangent (acoth) (#241)
stdlib-js/stdlib: lib/node_modules/@stdlib/math/base/special/acoth/docs/repl.txt

{{alias}}( x )
    Computes the inverse hyperbolic cotangent of a number.

>     If `x >= -1 && x <= 1`, the function returns `NaN`.

Reviewer: ```suggestion
    If provided a value on the interval [-1, 1], the function returns `NaN`.
```

    Parameters
    ----------
    x: number
        Input value.

    Returns
    -------

>     y: number|out

Reviewer: ```suggestion
    y: number
```

Author: I thought this was causing my linting error check because the function can return a number or NaN. I added it becasue the REPL documentation provides a case with multiple output values. Is `out` in this case the wrong return type or it should return just one type?

Reviewer: `NaN` is a number value, despite its "not-a-number" name. More specifically, it is a particular set of bit sequences which do not have a decimal representation when read from a value stored in double-precision floating-point format. Accordingly, even if a function returns `NaN`, using the `number` type is still appropriate.

Additionally, in this case, `out` is not a value type, so would not be appropriate in any context for parameter or return value types in REPL help documentation or in JSDoc type annotations.

You are correct, some functions in the project return values of different "types", but this is not one of those cases. :)

        Inverse hyperbolic cotangent (in radians).

    Examples
    --------
    > var y = {{alias}}( 1.0 )

>     0

Reviewer: ```suggestion
    0.0
```

I believe, based on the documentation, that this should be `NaN`, not `0.0`.

Author: NaN is for values in between -1 and 1. for these exact values, the javascript implementation return Infinity and I set this to 0 so it is more consistent with other functions.

Reviewer: For clarity, set to `0` to be consistent with what functions?
